src/formatBinaryPrefixAuto.js

```diff
@@ -0,0 +1,45 @@
+export var binaryPrefixExponent;
+
+export default function(x, p) {
+  var binaryExponent = 0;
+
+  while (Math.round(x) >= 1024 && binaryExponent < 80) {
```

Comment:
Is there a closed form for doing this, rather than needing a loop?

Reply:
Also it looks like this will hang if you try to format Infinity, but I did test it.

Reply:
I didn’t investigate any closed forms. I wouldn’t be surprised if one existed using the power of 2. But this loop will run at most 8 times as far as I can tell, even for Infinity.

Reply:
I added a test that confirms that infinity does not hang indefinitely. However if someone can point out a closed form, it would probably be preferable.

Reply:
That sounds right; I was too hasty reading the code. Thanks.

Reply:
Wouldn't it be possible to use `Math.log2` here?

(or the IE-compatible equivalent `Math.log(number) / Math.log(2)`)